README.md

```diff
@@ -1,63 +1,3280 @@
+`/var/run/resolv/interface/NetworkManager` file.
+
+The DNSMasq service runs at local address `127.0.0.2`. This is used, because
+it allows customer application services to provide their own DNS if required
+and therefore does not clash with them.
+
+By default, the external name servers used are the Google primary and secondary
+at `8.8.8.8` and  `8.8.4.4`. However, these can be overridden by modifying the
+`/mnt/boot/config.json` file and adding a `dnsServers` property, with a comma
+separated list of the IP addresses of the nameservers to use (see
+[the docs](https://github.com/balena-os/meta-balena#dnsservers) for more
+information).
+
+In this example, we're not going to use `balena ssh` for
+accessing the device as we'll be doing things to the system that include making
+it drop off the VPN (and at this point our shell would be terminated). Instead,
+we'll use `ssh` from your command line. Ensure you know the local IP
+address of the debug device (or use `balena scan` to find the hostname of your
+device), and SSH into it like this (where `192.168.1.173` is the IP address of
+your device, or `<host>.local` name):
+```
+$ ssh -p 22222 root@192.168.1.173
+The authenticity of host '[192.168.1.173]:22222 ([192.168.1.173]:22222)' can't be established.
+ECDSA key fingerprint is SHA256:xr1DjpoK8Ga6J318INrxP7bQbtALX8+//QlzoASbgWY.
+Are you sure you want to continue connecting (yes/no)? yes
+Warning: Permanently added '[192.168.1.173]:22222' (ECDSA) to the list of known hosts.
+root@f34c2e9:~#
+```
+
+We're going to modify the DNS servers to point at one that doesn't exist,
+just to show what happens. SSH into your device as above, then run the
+following:
+```
+root@51cc514:~# jq '.dnsServers = "1.2.3.4"' /mnt/boot/config.json > /mnt/data/config.json && mv /mnt/data/config.json /mnt/boot/config.json
+root@51cc514:~# mount -o remount,rw /
+root@51cc514:~# mv /etc/resolv.dnsmasq /etc/resolv.dnsmasq.moved
+root@51cc514:~# sync
+root@51cc514:~# reboot
+```
+This will move the default DNSMasq resolver config file, so that it's not
+picked up. Additionally, it will modify the configuration to set the nameserver
+to use as `1.2.3.4`. As this isn't a valid nameserver, nothing will get the
+right address to make connections. Note that usually, remounting the root FS
+as writable is a very risky move, and should not be carried out without good
+reason!
+
+After a while, once the device has rebooted, SSH back into the device, and look
+at the `dnsmasq.service` unit:
+```
+root@51cc514:~# systemctl status dnsmasq
+Warning: The unit file, source configuration file or drop-ins of dnsmasq.service changed on disk. Run 'systemctl daemon-reload' to reload units.
+● dnsmasq.service - DNS forwarder and DHCP server
+   Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled; vendor preset: enabled)
+  Drop-In: /etc/systemd/system/dnsmasq.service.d
+           └─dnsmasq.conf
+   Active: active (running) since Wed 2019-12-04 17:46:5
```

Comment:
there is also the case were users are logging/writting into their containers but not into volumes, so the actual container writeable layer baloons. This can be checked using `balena system df` and compared to the balena image sizes.

Reply:
Again, I think this is the space section, but need some context. :(

Reply:
Yeah this was just a general comment that we may want to talk about how users also write into their containers rather than volumes ( as in this section https://github.com/balena-io/debugging-masterclass/blob/e5292ab3bfb6fa51ed17299960ff422fb05bc35d/README.md#1112-customer-data ) as i have seen this a fair few times and users containers bloat massively because the write logs into the writteable layer

Reply:
Written an appropriate section.